src/vs/workbench/contrib/preferences/browser/settingsTreeModels.ts

```diff
@@ -288,6 +289,32 @@ export class SettingsTreeSettingElement extends SettingsTreeElement {
 
 		return Array.from(extensionFilters).some(extensionId => extensionId.toLowerCase() === this.setting.extensionInfo!.id.toLowerCase());
 	}
+
+	matchesAnyFeature(featureFilters?: Set<string>): boolean {
```

Comment:
This code has to run for every setting, so hundreds of times, we should try to do as much work up front as possible. Like if we can figure out what the feature is once, then apply its patterns to every setting quickly.

Reply:
where would I do this pre-processing?

Reply:
How about in the place where `matchesAnyFeature` is called from?

Reply:
yep. on it.